dotCMS/src/main/java/com/dotcms/publisher/bundle/business/BundleAPIImpl.java

```diff
@@ -82,7 +113,231 @@ public Bundle getBundleById(String id) throws DotDataException {
 	@Override
 	public void deleteBundle(String id) throws DotDataException {
 		bundleFactory.deleteBundle(id);
+	}
+
+	@WrapInTransaction
+	private void internalDeleteBundleAndDependencies (final String bundleId, final User user) throws DotDataException { // todo: make this batcheable
+
+		this.validateBundleDeletePermission(user, bundleId);
+
+		Logger.info(this, "Removing bundle: " + bundleId +
+				" and dependencies by: " + user.getUserId());
+
+		try {
+
+			Logger.info(this, "Removing audit status for a bundle: " + bundleId);
+			this.publishAuditAPI.deletePublishAuditStatus(bundleId);
+		} catch (DotPublisherException e) {
+
+			throw new DotDataException(e);
+		}
+
+		Logger.info(this, "Removing all pushed assets for a bundle: " + bundleId);
+		this.pushedAssetsAPI.deletePushedAssetsByBundleId(bundleId);
+
+		Logger.info(this, "Removing all assets from bundle: " + bundleId);
+		this.bundleFactory.deleteAllAssetsFromBundle(bundleId);
+
+		Logger.info(this, "Removing environments from bundle: " + bundleId);
+		this.bundleFactory.deleteBundleEnvironmentByBundle(bundleId);
+
+		Logger.info(this, "Removing bundle: " + bundleId);
+		this.deleteBundle(bundleId);
+	}
+
+	@WrapInTransaction
+	@Override
+	public void deleteBundleAndDependencies(final String bundleId, final User user) throws DotDataException {
+
+		final Bundle bundle = this.getBundleById(bundleId);
+		if (null != bundle) {
+
+			this.internalDeleteBundleAndDependencies(bundleId, user);
+		} else {
+			Logger.error(this,"The bundle id: " + bundleId + " does not exists");
+			throw new NotFoundInDbException("The bundle id: " + bundleId + " does not exists");
+		}
+	}
+
+	private void validateBundleDeletePermission(final User user, final String bundleId) throws DotDataException {
+
+		//This is meanwhile we make bundles permissionable, if the user is not an admin only the owner of the bundle can delete it
+		if(!userAPI.isCMSAdmin(user)){
+			if(!getBundleById(bundleId).getOwner().equals(user.getUserId())){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4502981)